GVRf/Framework/framework/src/main/java/org/gearvrf/GVRAssetLoader.java

```diff
@@ -45,23 +43,137 @@
 import org.gearvrf.scene_objects.GVRModelSceneObject;
 import org.gearvrf.utility.FileNameUtils;
 import org.gearvrf.utility.Log;
-
 import android.content.Context;
 import android.content.res.AssetManager;
 import android.os.Environment;
 import org.gearvrf.jassimp.AiColor;
+
 /**
- * {@link GVRImporter} provides methods for importing 3D models and making them
+ * {@link GVRAssetLoader} provides methods for importing 3D models and making them
  * available through instances of {@link GVRAssimpImporter}.
  * <p>
  * Supports importing models from an application's resources (both
  * {@code assets} and {@code res/raw}) and from directories on the device's SD
  * card that the application has permission to read.
  */
-final class GVRImporter {
-    private GVRImporter() {
-    }
+public final class GVRAssetLoader implements IEventReceiver {
```

Comment:
The asset loader is not a heavyweight class and doesn't keep a lot of stuff so there is no reason why you could not have multiple instances of it. I am not sure why we want to control the number of instances of this particular class. I plan to deprecate all the asset load functions on the context and have GVRAssetLoader provide the only API for loading. GVRContext has an asset loader for convenience - so there is always a way to load an asset from anywhere.